### PR TITLE
Added email validation on welcome and preferences screen

### PR DESCRIPTION
**Pull request by Pragya007**
Closes [#10209](https://github.com/desktop/desktop/issues/10209)

## Description

- Added email validation on welcome screen and preferences screen

### Screenshots

[gif1.zip](https://github.com/desktop/desktop/files/4927466/gif1.zip)

## Release notes

Notes:

### CODE REVIEW

File: app/src/ui/lib/configure-git-user.tsx [hunk 1]
```diff
@@ -185,6 +185,7 @@ export class ConfigureGitUser extends React.Component<
           />
 
           <TextBox
+            type="email"
```
Comment (SecretRose7) [inline app/src/ui/lib/configure-git-user.tsx:188]: GoldRoses.GR@gmail.com

File: app/src/ui/preferences/git.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ export class Git extends React.Component<IGitProps, {}> {
         </Row>
         <Row>
           <TextBox
+            type="email"
```
Comment (SecretRose7) [inline app/src/ui/preferences/git.tsx:27]: `GitRoses.GR@gmail.com`

File: app/src/ui/lib/configure-git-user.tsx (hunk 1 above)
Comment (rami0r) [reply app/src/ui/lib/configure-git-user.tsx:188]: 58536626+rami-HP@users.noreply.github.com

File: app/src/ui/preferences/git.tsx (hunk 2 above)
Comment (rami0r) [reply app/src/ui/preferences/git.tsx:27]: 58536626+rami-HP@users.noreply.github.com

**Review by rami-HP — COMMENTED** [review]
Duplicate of #

File: app/styles/ui/_text-box.scss [hunk 3]
```diff
@@ -24,6 +24,10 @@
     margin-bottom: var(--spacing-third);
   }
 
+  :invalid{
+    border-color: $red-500;
```
Comment (niik) [inline app/styles/ui/_text-box.scss:28]: Can we use our CSS variable for errors here, that way individual themes can modify it.

```suggestion
    border-color: var(--error-color);
```

File: app/styles/ui/_text-box.scss [hunk 4]
```diff
@@ -24,6 +24,10 @@
     margin-bottom: var(--spacing-third);
   }
 
+  :invalid{
```
Comment (niik) [inline app/styles/ui/_text-box.scss:27]: Missing a little space here to conform to our conventions

```suggestion
  :invalid {
```

**Review by niik — CHANGES_REQUESTED** [review]
Hey @Pragya007, thanks so much for working on this!

These changes look great, I had two small requests for changes that I'd like for you to look at but after than I think we should be able to get this merged and into the next release.

**Comment by Pragya007** [comment]
> Hey @Pragya007, thanks so much for working on this!
> 
> These changes look great, I had two small requests for changes that I'd like for you to look at but after than I think we should be able to get this merged and into the next release.

@niik  Sure, I will make the necessary changes.

**Review by niik — APPROVED** [review]
Thank you so much for working on this @Pragya007! Gonna try to get this into the next production release towards the end of the month

**Pull request merged**